[Bug]: Failed upgrade from v0.7.9 to v0.7.10

### 🐛 Describe the Bug

The latest release fails during an upgrade

### 🔍 Steps to Reproduce

Bump docker compose version of ztnet to v0.7.10 from v0.7.9

### 🌐 Browser

Other

### 🔧 Deployment Type

- [x] Docker
- [ ] Standalone

### ✨ Expected Behavior

Upgrade completes and ztnet comes up

### 📋 ZTNET Logs

```shell
ztnet     | Failed to load config file "/app/prisma.config.ts" as a TypeScript/JavaScript module. Error: Error: Cannot find module 'dotenv/config'
ztnet     | Require stack:
ztnet     | - /app/prisma.config.ts
ztnet     | An error occurred. Exiting...
ztnet     | Creating .env file...
ztnet     | Applying migrations to the database...
ztnet     | Failed to load config file "/app/prisma.config.ts" as a TypeScript/JavaScript module. Error: Error: Cannot find module 'dotenv/config'
ztnet     | Require stack:
ztnet     | - /app/prisma.config.ts
ztnet     | An error occurred. Exiting...
ztnet     | Creating .env file...
ztnet     | Applying migrations to the database...
ztnet     | Failed to load config file "/app/prisma.config.ts" as a TypeScript/JavaScript module. Error: Error: Cannot find module 'dotenv/config'
ztnet     | Require stack:
ztnet     | - /app/prisma.config.ts
ztnet     | An error occurred. Exiting...
ztnet exited with code 1 (restarting)
postgres  | 2025-12-05 05:29:36.479 UTC [22] LOG:  checkpoint starting: time
postgres  | 2025-12-05 05:29:37.715 UTC [22] LOG:  checkpoint complete: wrote 13 buffers (0.1%); 0 WAL file(s) added, 0 removed, 0 recycled; write=1.206 s, sync=0.007 s, total=1.237 s; sync files=3, longest=0.005 s, average=0.003 s; distance=90 kB, estimate=91 kB
ztnet     | Creating .env file...
ztnet     | Applying migrations to the database...
ztnet     | Failed to load config file "/app/prisma.config.ts" as a TypeScript/JavaScript module. Error: Error: Cannot find module 'dotenv/config'
ztnet     | Require stack:
ztnet     | - /app/prisma.config.ts
ztnet     | An error occurred. Exiting...
ztnet exited with code 1 (restarting)
ztnet     | Creating .env file...
ztnet     | Applying migrations to the database...
ztnet     | Failed to load config file "/app/prisma.config.ts" as a TypeScript/JavaScript module. Error: Error: Cannot find module 'dotenv/config'
ztnet     | Require stack:
ztnet     | - /app/prisma.config.ts
ztnet     | An error occurred. Exiting...
ztnet exited with code 1 (restarting)
```

### 🖼 Screenshots

_No response_

--- a/.devcontainer/init-cmd.sh
+++ b/.devcontainer/init-cmd.sh
@@ -2,11 +2,8 @@
 
 set -e
 
-# Create .env file
-# cat << EOF > .env
-# DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}?schema=public
-# ZT_ADDR=${ZT_ADDR}
-# EOF
+# Export DATABASE_URL for Prisma (prisma.config.ts skips .env loading)
+export DATABASE_URL="postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}?schema=public"
 
 until PGPASSWORD=$POSTGRES_PASSWORD psql -h "$POSTGRES_HOST" -U "$POSTGRES_USER" -c '\q'; do
   >&2 echo "Postgres is unavailable - sleeping"

--- a/src/utils/ztApi.ts
+++ b/src/utils/ztApi.ts
@@ -508,20 +508,29 @@ export const local_network_and_members = async (
 	const { headers, localControllerUrl } = await getOptions(ctx, false);
 
 	try {
-		const members = await network_members(ctx, nwid, false);
 		const network = await getData<NetworkEntity>(
 			`${localControllerUrl}/controller/network/${nwid}`,
 			headers,
 		);
+		const members = await network_members(ctx, nwid, false);
 
-		// Fetch member details sequentially to avoid overwhelming the ZeroTier controller
+		// Fetch member details in batches to avoid overwhelming the ZeroTier controller
+		// Batch size of 5 balances performance with reliability on resource-constrained systems
+		const BATCH_SIZE = 5;
+		const memberEntries = Object.entries(members);
 		const membersArr: MemberEntity[] = [];
-		for (const [memberId] of Object.entries(members)) {
-			const memberDetails = await getData<MemberEntity>(
-				`${localControllerUrl}/controller/network/${nwid}/member/${memberId}`,
-				headers,
+
+		for (let i = 0; i < memberEntries.length; i += BATCH_SIZE) {
+			const batch = memberEntries.slice(i, i + BATCH_SIZE);
+			const batchResults = await Promise.all(
+				batch.map(async ([memberId]) => {
+					return await getData<MemberEntity>(
+						`${localControllerUrl}/controller/network/${nwid}/member/${memberId}`,
+						headers,
+					);
+				}),
 			);
-			membersArr.push(memberDetails);
+			membersArr.push(...batchResults);
 		}
 
 		return {

--- a/src/utils/ztApi.ts
+++ b/src/utils/ztApi.ts
@@ -508,24 +508,21 @@ export const local_network_and_members = async (
 	const { headers, localControllerUrl } = await getOptions(ctx, false);
 
 	try {
-		// Fetch network details first, then members sequentially
-		// This avoids hitting concurrent connection limits when multiple networks are fetched in parallel
+		const members = await network_members(ctx, nwid, false);
 		const network = await getData<NetworkEntity>(
 			`${localControllerUrl}/controller/network/${nwid}`,
 			headers,
 		);
-		const members = await network_members(ctx, nwid, false);
 
-		// PERFORMANCE OPTIMIZATION: Fetch all member details in parallel instead of sequentially
-		// This significantly reduces total request time from O(n) to O(1) for network latency
-		const membersArr: MemberEntity[] = await Promise.all(
-			Object.entries(members).map(async ([memberId]) => {
-				return await getData<MemberEntity>(
-					`${localControllerUrl}/controller/network/${nwid}/member/${memberId}`,
-					headers,
-				);
-			}),
-		);
+		// Fetch member details sequentially to avoid overwhelming the ZeroTier controller
+		const membersArr: MemberEntity[] = [];
+		for (const [memberId] of Object.entries(members)) {
+			const memberDetails = await getData<MemberEntity>(
+				`${localControllerUrl}/controller/network/${nwid}/member/${memberId}`,
+				headers,
+			);
+			membersArr.push(memberDetails);
+		}
 
 		return {
 			network,

--- a/src/utils/ztApi.ts
+++ b/src/utils/ztApi.ts
@@ -515,7 +515,6 @@ export const local_network_and_members = async (
 		const members = await network_members(ctx, nwid, false);
 
 		// Fetch member details in batches to avoid overwhelming the ZeroTier controller
-		// Batch size of 5 balances performance with reliability on resource-constrained systems
 		const BATCH_SIZE = 5;
 		const memberEntries = Object.entries(members);
 		const membersArr: MemberEntity[] = [];

